### PR TITLE
IR cleanup (part 1)

### PR DESCRIPTION
**Pull request by gramalingam**
Cleanup the IR and IR builder: add type annotations, rename.

### CODE REVIEW

File: onnxscript/irbuilder.py [hunk 1]
```diff
@@ -344,35 +353,25 @@ def to_graph_proto(self, use_default_type: bool = True):
...
 
-            Default values for attributes are introduced in onnx==1.13.0.
-            If an earlier version of onnx is installed, it ignores the default
-            values of the function arguments.
+        Note: Default values for attributes are an experimental feature in ONNX.
+        Conversion ignores default values for attributes if the ONNX version installed
+        doesn't support it.
         """
         opsets = self.get_opset_import()
         if domain != "":
```
Comment (justinchuby) [inline onnxscript/irbuilder.py:377]: domian can be a type other string right

**Review by justinchuby — APPROVED** [review]

File: onnxscript/irbuilder.py (hunk 1 above)
Comment (gramalingam) [reply onnxscript/irbuilder.py:377]: I don't understand the question. The type is string. (But I don't know why we have this parameter, I think we should eventually eliminate it. I will look into it.)

Comment (gramalingam) [reply onnxscript/irbuilder.py:377]: Oh, you are referring to the next line, I see now. Anyway, let me see if I can delete this parameter.

File: onnxscript/irbuilder.py [hunk 2]
```diff
@@ -328,17 +329,17 @@ def to_graph_and_functions(
         Returns:
             a pair of a :class:`onnx.GraphProto` and list of :class:`onnx.FunctionProto`
         """
-        sub_functions = {}
+        called_functions: Dict[str, onnx.FunctionProto] = {}
```
Comment (justinchuby) [inline onnxscript/irbuilder.py:332]:
```suggestion
        called_functions: dict[str, onnx.FunctionProto] = {}
```

nit

File: onnxscript/type_annotation.py [hunk 3]
```diff
@@ -62,9 +69,41 @@ def is_attr(pytype: type):
...
 def is_valid(typeinfo):
     return is_attr(typeinfo) or is_tensor(typeinfo)
+
+
+def validate(typeinfo):
+    if not is_valid(typeinfo):
+        raise ValueError(f"Unsupported type annotation {typeinfo}")
+
+
+def get_return_types(typeinfo: type | typing.Sequence[type]) -> typing.Sequence[type]:
```
Comment (justinchuby) [inline onnxscript/type_annotation.py:91]:
```suggestion
def get_return_types(typeinfo: type | Sequence[type]) -> Sequence[type]:
```

nit

Comment (gramalingam) [reply onnxscript/type_annotation.py:84]: Currently importing only `typing` in this file (due to 3.8 and 3.7 version differences for get_origin/get_args).

**Pull request merged**